src/lib/sub-bots/sub-bots.js

```diff
@@ -45,26 +46,25 @@ module.exports = {
     passBot: function() {
         crashHandler.logEvent(TAG, 'passBot');
         return new Promise(function(resolve, reject) {
-            // Probably best not to hit this, should be checked by what's calling it for better handling
-            if (!config.getConfig().subBots) {
-                logger.info(TAG, 'SubBots feature called, but disabled or no subBots on file, rejecting');
-                reject('The sub bot feature is disabled or there are no subBots on file');
-                return;
-            }
 
-            if (currentBots >= config.getConfig().subBotLimit) {
+            if (currentBots >= config.get('subBotLimit')) {
                 logger.info(TAG, 'SubBot requested but maximum number logged on');
                 reject('The maximum number of subBots are currently running');
                 return;
-            } else {
-                currentBots++;
```

Comment:
I believe this is an error in asynchronous logic. The current code marks a subbot as busy in the instant that it is requested and one is available. This prevents two requests attempting to be fulfilled when a second request is made before the first is fulfilled when only one sub-bot remains.

Discord message -> API -> our code and login request -> login success response ≈ 1-2s

Reply:
I didn't move it to it's original location. So now it will only increase after it finds a subbot which is not busy, which makes more sense.